d2compiler/compile.go

```diff
@@ -584,6 +584,16 @@ func (c *compiler) compileReserved(attrs *d2graph.Attributes, f *d2ir.Field) {
 		attrs.Link = &d2graph.Scalar{}
 		attrs.Link.Value = scalar.ScalarString()
 		attrs.Link.MapKey = f.LastPrimaryKey()
+
+		if attrs.Label.Value == "" {
+			attrs.Label.Value = attrs.Link.Value
+		} else {
+			// this works because the Label attribute is set before the Link attribute
+			u, err := url.ParseRequestURI(attrs.Label.Value)
+			if err == nil && u.Host != "" {
+				c.errorf(f.Primary_.Value, "Label cannot be set to URL when link is also set (for security)")
+			}
+		}
```

Comment:
actually here you can't tell if it's a field for an edge or a shape. 

Just after `compileReserved` here then: https://github.com/alixander/d2/blob/94d5ab5d2ec4be33e4227bc767e2e87068ce00ef/d2compiler/compile.go#L832